test/conftest.py

```diff
@@ -1,29 +1,72 @@
+import time
 import pytest
 import torch
 from multiprocessing import Process
 import builtins
 
 import syft
 from syft import TorchHook
+from syft.workers import WebsocketClientWorker
+from syft.workers import WebsocketServerWorker
+
+
+def _start_proc(participant, dataset: str = None, **kwargs):  # pragma: no cover
+    """Helper function for spinning up a websocket participant."""
+
+    def target():
+        server = participant(**kwargs)
+        if dataset is not None:
+            data, key = dataset
+            server.add_dataset(data, key=key)
+        server.start()
+
+    p = Process(target=target)
+    p.start()
+    return p
+
+
+def instantiate_websocket_client_worker(max_tries=5, sleep_time=0.1, **kwargs):  # pragma: no cover
+    """Helper function to instantiate the websocket client.
+
+    If a connection is refused, we wait a bit (`sleep_time` seconds) and try again.
+    After `max_tries` failed tries, a ConnectionRefusedError is raised.
+    """
+    retry_counter = 0
+    connection_open = False
+    while not connection_open:
+        try:
+            local_worker = WebsocketClientWorker(**kwargs)
+            connection_open = True
+        except ConnectionRefusedError as e:
+            if retry_counter < max_tries:
+                retry_counter += 1
+                time.sleep(sleep_time)
+            else:
+                raise e
+    return local_worker
 
 
 @pytest.fixture()
 def start_proc():  # pragma: no cover
-    """ helper function for spinning up a websocket participant """
+    return _start_proc
 
-    def _start_proc(participant, dataset: str = None, **kwargs):
-        def target():
-            server = participant(**kwargs)
-            if dataset is not None:
-                data, key = dataset
-                server.add_dataset(data, key=key)
-            server.start()
 
-        p = Process(target=target)
-        p.start()
-        return p
+@pytest.fixture()
+def start_remote_worker():  # pragma: no cover
+    """Helper function for starting a websocket worker."""
 
-    return _start_proc
+    def _start_remote_worker(
+        id, hook, dataset: str = None, host="localhost", port=8768, max_tries=5, sleep_time=0.01
+    ):
+        kwargs = {"id": id, "host": host, "port": port, "hook": hook}
+        server = _start_proc(WebsocketServerWorker, dataset=dataset, **kwargs)
+        remote_worker = instantiate_websocket_client_worker(
```

Comment:
I would prefer another name for remote_worker (as remote_proxy). As this remote_worker instance is the local proxy of the remote websocket server.